src/Microsoft.Azure.ServiceBus/Primitives/ExceptionReceivedContext.cs

```diff
@@ -0,0 +1,29 @@
+namespace Microsoft.Azure.ServiceBus.Primitives
+{
+    using System;
+
+    /// <summary>Context provided for <see cref="ExceptionReceivedEventArgs"/> exception raised by the client.</summary>
+    public class ExceptionReceivedContext
+    {
+        /// <summary>Initializes a new instance of the <see cref="ExceptionReceivedContext" /> class.</summary>
+        /// <param name="action">The action associated with the exception.</param>
+        /// <param name="endpoint">The endpoint associated with the exception.</param>
+        /// <param name="entityPath">The entity path associated with the exception.</param>
+        public ExceptionReceivedContext(string action, string endpoint, string entityPath)

```

Comment:
If someone want to perform testing and needs to construct the object to mock behavior, `internal` will get in a way. Do you still think it should be internal?